usaspending_api/references/views_v2.py

```diff
@@ -24,17 +25,53 @@ def get(self, request, pk, format=None):
         queryset = SubmissionAttributes.objects.all()
         queryset = queryset.filter(cgac_code=toptier_agency.cgac_code)
 
-        # get the most up to date fy
-        queryset = queryset.order_by('-reporting_fiscal_year')
+        # get the most up to date fy and quarter
+        queryset = queryset.order_by('-reporting_fiscal_year', '-reporting_fiscal_quarter')
         queryset = queryset.annotate(
-            fiscal_year=F('reporting_fiscal_year')
+            fiscal_year=F('reporting_fiscal_year'),
+            fiscal_quarter=F('reporting_fiscal_quarter')
+        )
+        submission = queryset.first()
+        if submission is None:
+            return Response(response)
+        active_fiscal_year = submission.reporting_fiscal_year
+        active_fiscal_quarter = submission.fiscal_quarter
+
+        # using final_objects below ensures that we're only pulling the latest
+        # set of financial information for each fiscal year
+        queryset = AppropriationAccountBalances.final_objects.all()
+        # get the incoming agency's toptier agency, because that's what we'll
+        # need to filter on
+        # (used filter() instead of get() b/c we likely don't want to raise an
+        # error on a bad agency id)
+        queryset = queryset.filter(
+            submission__reporting_fiscal_year=active_fiscal_year,
+            treasury_account_identifier__funding_toptier_agency=toptier_agency
         )
+        queryset = queryset.annotate(
+            budget_authority_amount=Sum('budget_authority_available_amount_total_cpe'),
+            obligated_amount=Sum('obligations_incurred_total_by_tas_cpe'),
+            outlay_amount=Sum('gross_outlay_amount_by_tas_cpe'))
+
         submission = queryset.first()
         if submission is None:
             return Response(response)
-        active_fiscal_year = submission.fiscal_year
+
+        # get the overall total government budget authority (to craft a budget authority percentage)
+        queryset2 = OverallTotals.objects.all()
+        queryset2 = queryset2.filter(fiscal_year=active_fiscal_year)
+
+        submission2 = queryset2.first()
 
         # craft response
-        response['results']['agency_name'] = toptier_agency.name
+        response['results']['total_budget_authority_amount'] = str(submission2.total_budget_authority)
```

Comment:
Here we could make it look a bit better when compiling the response by doing something like:
```
response['results'] = { 'total_budget_authority_amount': str(submission2.total_budget_authority),
                                      'budget_authority_amount': str(submission.budget_authority_amount),
                                      ...
                                     }
```

With better spacing than my example 😄